infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InSnapshotSyncState.java

```diff
@@ -75,20 +75,16 @@ public InSnapshotSyncState(LogReplicationFSM logReplicationFSM, SnapshotSender s
     public LogReplicationState processEvent(LogReplicationEvent event) throws IllegalTransitionException {
         switch (event.getType()) {
             case SNAPSHOT_SYNC_REQUEST:
-                /*
-                 Cancel ongoing snapshot sync, if it is still in progress.
-                 */
+
+                // Cancel ongoing snapshot sync, if it is still in progress.
                 setForcedSnapshotSync(event.getMetadata().isForcedSnapshotSync());
                 String cancelCause = forcedSnapshotSync ? "incoming forced snapshot sync." : "another snapshot sync request.";
                 cancelSnapshotSync(cancelCause);
 
-                /*
-                 Set the id of the new snapshot sync request causing the transition.
 
-                 This will be taken onEntry of this state to initiate a snapshot send
-                 for this given request.
-                 */
-                setTransitionEventId(event.getEventId());
+                // Set the id of the new snapshot sync request causing the transition.
+                // This will be taken onEntry of this state to initiate a snapshot send for this given request.
+                setTransitionEventId(event.getMetadata().getRequestId());
```

Comment:
from the name and comment for 'transitionEventId', it is used to track the event which caused this transition.  How is it useful to change it to the request id?

Reply:
we can change the name to just transitionId if its more meaningful..
Essentially there is no use of tracking which event caused us to transition to the current state. Instead its useful to understand which sync-id is being processed by the current state. In this way we can stop processing events that gets generated by stale syncs, and only transition if the sync-id from the event is the same as the ID of the current state.

Reply:
Makes sense.  Can you change the explanation for 'transitionEventId' accordingly so that it is not confusing?